lib/tasks/test.rake

```diff
@@ -9,86 +9,197 @@ namespace :test do
+      !file.end_with?('.xml')
+    end
   end
 
-  def hets_out_file_for(ontology_file)
-    basename = File.basename(ontology_file).sub(/\.[^.]+$/, '.xml')
-    File.join('spec/fixtures/ontologies/hets-out/', basename)
+  def prove_files
+    all_files_beneath('spec/fixtures/ontologies/prove')
+  end
+
+  def absolute_filepath(file)
+    Rails.root.join(file)
   end
 
-  def on_outdated_files(files, &block)
-    files.each do |file|
-      hets_file = hets_out_file_for(file)
-      out_of_date = !FileUtils.uptodate?(hets_file, Array(hets_path))
-      block.call(file) if out_of_date
+  def cassette_file(file)
+    # remove spec/fixtures/ontologies/ for cassette name
+    cassette_filepath = file.split('/')[3..-1].join('/')
+  end
+
+  def hets_cassette_dir(subdir)
+    File.join('hets-out', subdir)
+  end
+
+  def cassette_path_in_fixtures(subdir, file)
+    File.join(hets_cassette_dir(subdir), cassette_file(file))
+  end
+
+  def recorded_file(subdir, file)
+    base = file.split('.')[0..-2].join('.')
+    old_extension = File.extname(file)[1..-1]
+    file = "#{base}_#{old_extension}.yml"
+    File.join('spec', 'fixtures', 'vcr', cassette_path_in_fixtures(subdir, file))
+  end
+
+  def outdated_cassettes(files, subdir)
+    files.select do |file|
+      cassette = recorded_file(subdir, file)
+      !FileUtils.uptodate?(cassette, Array(HETS_PATH))
     end
   end
 
-  def hets_args
-    YAML.load(File.open('config/hets.yml'))['cmd_line_options']
+  def on_outdated_cassettes(files, subdir, &block)
+    outdated_cassettes(files, subdir).each do |file|
+      block.call(file, subdir)
+    end
   end
 
-  def perform_hets_on(file)
-    args = hets_args << '-O spec/fixtures/ontologies/hets-out'
-    system("hets #{args.join(' ')} #{file}")
+  def call_hets_dg(file, subdir)
+    puts "Calling hets/dg on #{file.inspect}"
+    hets_api_options = "#{HETS_API_OPTIONS}/full-signatures/full-theories"
+    escaped_iri = Rack::Utils.escape_path("file://#{absolute_filepath(file)}")
+    hets_iri = "#{HETS_BASE_IRI}/dg/#{escaped_iri}#{hets_api_options}"
+
+    FileUtils.rm_f(recorded_file(subdir, file))
+    VCR.use_cassette(cassette_path_in_fixtures(subdir, file)) do
+      Net::HTTP.get_response(URI(hets_iri))
+    end
   end
 
-  def ontology_files
-    globbed_files = Dir.glob('spec/fixtures/ontologies/**/*')
-    globbed_files.select do |file|
-      !file.end_with?('proof.json') &&
-      !file.end_with?('xml') &&
-      !File.directory?(file)
+  def call_hets_provers(file, subdir)
+    puts "Calling hets/provers on #{file.inspect}"
+    options = 'format=json'
+    escaped_iri = Rack::Utils.escape_path("file://#{absolute_filepath(file)}")
+    hets_iri = "#{HETS_BASE_IRI}/provers/#{escaped_iri}#{HETS_API_OPTIONS}"
+    hets_iri << "?#{options}"
+
+    FileUtils.rm_f(recorded_file(subdir, file))
+    VCR.use_cassette(cassette_path_in_fixtures(subdir, file)) do
+      Net::HTTP.get_response(URI(hets_iri))
```

Comment:
maybe part of the code can be abstracted here, since the call_hets_* methods are quite similar. Cf. also lib/hets/*_caller.rb